### ISSUE TITLE
Not letting me add Int as a type?

### ISSUE THREAD

**Issue by MarcGodard** [prisma/prisma#14389]
### Bug description

I added a single value into a model of type `Int` and when I go to push it to the database, I get the following error.

I had just updated the version to the newest (was on 4.0.0 before). I even tried on an empty clean database.

```
Error: db error: ERROR: operator is not unique: smallint[] @> smallint[]
HINT: Could not choose a best candidate operator. You might need to add explicit type casts.
   0: sql_migration_connector::flavour::postgres::connection::describe_schema
             at migration-engine/connectors/sql-migration-connector/src/flavour/postgres/connection.rs:30
   1: migration_core::state::SchemaPush
             at migration-engine/core/src/state.rs:349

```

### Prisma information

Added the opened Int, this model had another Int already. Also, I changed the variable name and it still caused the error.

```
model Redirects {
  id          String   @id @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  linkName    String   @unique
  status      Int      @default(307)
  location    String   @default("/")
  opened      Int      @default(0)
  cookie      Boolean
  cookieName  String?
  cookieValue String?
  createdAt   DateTime @default(now())
  updatedAt   DateTime @default(now())
}
```

### Environment & setup

- OS: Linux
- Database: PostgreSQL
- Node.js version: v16.14.0

### Prisma Version

```
prisma                  : 4.1.0
@prisma/client          : 4.1.0
Current platform        : debian-openssl-3.0.x
Query Engine (Node-API) : libquery-engine 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Migration Engine        : migration-engine-cli 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/migration-engine-debian-openssl-3.0.x)
Introspection Engine    : introspection-core 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/introspection-engine-debian-openssl-3.0.x)
Format Binary           : prisma-fmt 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-3.0.x)
Default Engines Hash    : 8d8414deb360336e4698a65aa45a1fbaf1ce13d8
Studio                  : 0.467.0
```

**Comment by MarcGodard** [comment]
I installed v 4.0 of prisma and no issues.

**Comment by tomhoule** [comment]
The operator ambiguity may be caused by the `intarray` module on postgres, do you have that enabled? (from https://stackoverflow.com/questions/39798993/gin-index-on-smallint-column-not-used-or-error-operator-is-not-unique) — there are solutions, we can look into these once we understand a bit more.

**Comment by MarcGodard** [comment]
@tomhoule I am using elephantsql.com

It says the version is: PostgreSQL version | 13.4 (Ubuntu 13.4-1.pgdg20.04+1)

**Comment by tomhoule** [comment]
No that sounds good, thanks for the information. I think the problem is with that extension, we should be able to fix this for Prisma 4.2.0.

**Comment by MarcGodard** [comment]
Looking forward to it.

**Comment by alexanderniebuhr** [comment]
I probably can confirm. We have installed the `intarray` extension and get the same issue.

**Comment by zaida04** [comment]
I can reproduce this issue on another elephantsql database, same as OP.

**Comment by abenhamdine** [comment]
@tomhoule are you aware that prisma is currently unusable with postgres with intarray installed (thus the vast majority of hosted postgres databases) because of this issue ? Do you plan to release a fix soon ?

**Comment by nasainsbury** [comment]
I had the same issue kinda. When running locally on postgres 12.5 it was fine, however when I tried migrating on a managed DB (render), for v12 (not sure of minor version), I get the above error. To test, I created a new DB on v14 and it was solved.

Not sure if this helps, likely not.

**Comment by tomhoule** [comment]
> The issue occurs with prisma@4.1.1 and even with prisma@4.0.0

Can you confirm that this happens with Prisma 4.0.0? That means our current way of thinking about this would be completely wrong, and make this an unknown bug. I am fairly certain the code change that triggers this issue went out with 4.1.0.

And yes, this is prioritized as a bug to fix for 4.2.0.

**Comment by tomhoule** [comment]
This issue should be fixed by https://github.com/prisma/prisma-engines/pull/3096 — it will be out in `prisma@dev` today and in the 4.2 release next week.
